dynawo/sources/Models/CPP/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp

```diff
@@ -372,57 +290,68 @@ namespace DYN {
-      jt.addTerm(UrYNum_ + rowOffset, - (P() * U2() - 2 * Ur() * (P() * Ur() + Q() * Ui())) / (U2() * U2()));
-      jt.addTerm(UiYNum_ + rowOffset, - (Q() * U2() - 2 * Ui() * (P() * Ur() + Q() * Ui())) / (U2() * U2()));
+      jt.addTerm(PYNum_ + rowOffset, -Ur / U2);
+      jt.addTerm(QYNum_ + rowOffset, -Ui / U2);
+      jt.addTerm(UrYNum_ + rowOffset, - (P * U2 - 2 * Ur * (P * Ur + Q * Ui)) / (U2 * U2));
+      jt.addTerm(UiYNum_ + rowOffset, - (Q * U2 - 2 * Ui * (P * Ur + Q * Ui)) / (U2 * U2));
       jt.changeCol();
       jt.addTerm(IiYNum_ + rowOffset, 1);
-      jt.addTerm(PYNum_ + rowOffset, -Ui() / U2());
-      jt.addTerm(QYNum_ + rowOffset, Ur() / U2());
-      jt.addTerm(UiYNum_ + rowOffset, - (P() * U2() - 2 * Ui() * (P() * Ui() - Q() * Ur())) / (U2() * U2()));
-      jt.addTerm(UrYNum_ + rowOffset, (Q() *  U2() + 2 * Ur() * (P() * Ui() - Q() * Ur())) / (U2() * U2()));
+      jt.addTerm(PYNum_ + rowOffset, -Ui / U2);
+      jt.addTerm(QYNum_ + rowOffset, Ur / U2);
+      jt.addTerm(UiYNum_ + rowOffset, - (P * U2 - 2 * Ui * (P * Ui - Q * Ur)) / (U2 * U2));
+      jt.addTerm(UrYNum_ + rowOffset, (Q *  U2 + 2 * Ur * (P * Ui - Q * Ur)) / (U2 * U2));
       jt.changeCol();
       jt.addTerm(PYNum_ + rowOffset, 1);
-      jt.addTerm(UrYNum_ + rowOffset, - P_dUr(Ur(), Ui(), U(), U2()));
-      jt.addTerm(UiYNum_ + rowOffset, - P_dUi(Ur(), Ui(), U(), U2()));
-      jt.addTerm(UfYNum_ + rowOffset, - P_dUf(U()));
+      jt.addTerm(UrYNum_ + rowOffset, - P_dUr);
+      jt.addTerm(UiYNum_ + rowOffset, - P_dUi);
+      jt.addTerm(UfYNum_ + rowOffset, - P_dUf);
       jt.changeCol();
       jt.addTerm(QYNum_ + rowOffset, 1);
-      jt.addTerm(UrYNum_ + rowOffset, - Q_dUr(Ur(), Ui(), U(), U2()));
-      jt.addTerm(UiYNum_ + rowOffset, - Q_dUi(Ur(), Ui(), U(), U2()));
-      jt.addTerm(UfYNum_ + rowOffset, - Q_dUf(U()));
+      jt.addTerm(UrYNum_ + rowOffset, - Q_dUr);
+      jt.addTerm(UiYNum_ + rowOffset, - Q_dUi);
+      jt.addTerm(UfYNum_ + rowOffset, - Q_dUf);
     }
   }
 
   void
   ModelLoadRestorativeWithLimits::evalG(const double& /*t*/) {
-    gLocal_[0] = (doubleNotEquals(UfRaw(), UMaxPu_) && UfRaw() > UMaxPu_) ? ROOT_UP : ROOT_DOWN;
-    gLocal_[1] = (doubleNotEquals(UfRaw(), UMinPu_) && UMinPu_ - UfRaw() > 0) ? ROOT_UP : ROOT_DOWN;
+    gLocal_[0] = (doubleNotEquals(yLocal_[UfRawYNum_], UMaxPu_) && yLocal_[UfRawYNum_] > UMaxPu_) ? ROOT_UP : ROOT_DOWN;
+    gLocal_[1] = (doubleNotEquals(yLocal_[UfRawYNum_], UMinPu_) && UMinPu_ - yLocal_[UfRawYNum_] > 0) ? ROOT_UP : ROOT_DOWN;
   }
 
   void
   ModelLoadRestorativeWithLimits::setGequations() {
-    if (getIsInitProcess())
-      return;
-    if (isConnected()) {
-      if (gLocal_[0] == ROOT_UP) {
-        gEquationIndex_[0] = std::string("UMaxPu limit reached by Load: ").append(staticId());
-      }
-      if (gLocal_[1] == ROOT_UP) {
-        gEquationIndex_[1] = std::string("UMinPu limit reached by Load: ").append(staticId());
-      }
+    if (gLocal_[0] == ROOT_UP) {
```

Comment:
Should not be protected, the description of the equation is always available

Reply:
ok, thanks